itest/tests/contextMenu.test.js

```diff
@@ -34,24 +61,63 @@ describe("Test search mods via right-clicks", () => {
     }
   })
 
-  beforeEach(async () => {
-    await writeSearch(app, "")
-    await startSearch(app)
-    await waitForResults(app)
+  stdTest("FilterEq ensure ingest", (done) => {
+    writeSearch(app, `* | count() by _path | sort _path`)
+      .then(async () => {
```

Comment:
I've observed this pattern in the stdTests. The first promise used uses the `.then` syntax, and the following promises use the `await` syntax within the first `then`. Finally, the first promise `catch`es the any error and calls our common error handler.

I'd guess that pattern is there so that we can catch all the errors without writing a try catch block. Is that correct?

I thought about a potential solution that would reduce some of the handleError boilerplate and would not need to switch between the different promise syntax.

```js
// Add the app to the stdTest args (needed to catch the errors)
function stdTest(desc, app, func, timeout) {
  test(desc, async (done) => {
    LOG.info(`Running test: ${descr}`)
    try {
      await func(done)
    } catch (err) {
      handleError(app, err, done)
    }
  }, timeout)
}
```

Now the code in the test can always use the await syntax and won't need to handle the error.

```js
stdTest("my test", app, async (done) => {
  await writeSearch(app, "*")
  await startSearch(app)
  await searchDisplay(app)
  ...
  done()
})
```

For your consideration, not a blocker.

Reply:
Appreciate the feedback. It's good you're starting to look about ways to improve integration tests. The feedback helps me be a better Javascript writer.

`stdTest` needs work for sure. I wrote it in early days before I even started using `async` / `await`.

`handleError` really should be used inside `stdTest`; I agree. The flow signature on `stdTest` also somewhat limits its usage.

Something else `stdTest` should do: get the prototype of a jest `test`. This doesn't work for example:
```javascript
stdTest.skip("my test", ....)
```